File: MosPosudit.UI/shared/lib/dtos/order/order_insert_request.dart
class OrderInsertRequest {
  final int userId;
  final DateTime startDate;
  final DateTime endDate;
  final bool termsAccepted;
  final List<OrderItemInsertRequest> orderItems;

  OrderInsertRequest({
    required this.userId,
    required this.startDate,
    required this.endDate,
    required this.termsAccepted,
    required this.orderItems,
  });

  Map<String, dynamic> toJson() => {
        'userId': userId,
        'startDate': startDate.toIso8601String(),
        'endDate': endDate.toIso8601String(),
        'termsAccepted': termsAccepted,
        'orderItems': orderItems.map((item) => item.toJson()).toList(),
      };
}

class OrderItemInsertRequest {
  final int toolId;
  final int quantity;

  OrderItemInsertRequest({
    required this.toolId,
    required this.quantity,
  });

  Map<String, dynamic> toJson() => {
        'toolId': toolId,
        'quantity': quantity,
      };
}

File: MosPosudit.Model/Requests/Order/OrderInsertRequest.cs
using System.ComponentModel.DataAnnotations;

namespace MosPosudit.Model.Requests.Order
{
    public class OrderInsertRequest
    {
        [Required(ErrorMessage = "User ID is required")]
        public int UserId { get; set; }

        [Required(ErrorMessage = "Start date is required")]
        public DateTime StartDate { get; set; }

        [Required(ErrorMessage = "End date is required")]
        public DateTime EndDate { get; set; }

        [Required(ErrorMessage = "Terms must be accepted")]
        public bool TermsAccepted { get; set; }

        [Required(ErrorMessage = "Order items are required")]
        [MinLength(1, ErrorMessage = "At least one order item is required")]
        public List<OrderItemInsertRequest> OrderItems { get; set; } = new List<OrderItemInsertRequest>();
    }
}

